utils/docker/Dockerfile.centos.7

```diff
@@ -12,13 +12,44 @@ LABEL maintainer="daos@daos.groups.io"
 # Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
 ARG CB0
 
+# Need to be able to use a local repository for packages
+ARG REPO_URL=""
+ARG REPO_EL7=""
+
+# Install EPEL and switch to dnf as it seems a bit faster.
+RUN yum clean all &&                                            \
+    yum -y install dnf epel-release yum-utils &&                \
+    yum clean all
+
+# If a local repository is supplied, then we should use it
+# and disable using the distro repositories.
+RUN if [ -n "$REPO_URL" ]; then                                 \
+      MY_REPO="${REPO_URL}${REPO_EL7}/";                        \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
+      echo -e "[${MY_NAME}]\n\
+name=created from ${MY_REPO}\n\
+baseurl=${MY_REPO}\n\
+enabled=1\n\
+repo_gpgcheck=0\n\
+exclude=fuse-3* fuse-devel-3* fuse-libs-3* fuse-debuginfo-3* ompi*\n\
+gpgcheck=0\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
+      yum clean all;                                            \
+      yum-config-manager --assumeyes --quiet                    \
+                         --disable base epel extra updates;     \
+    fi;                                                         \
+    sed "s/^mirrorlist_expire=0*/mirrorlist_expire=99999999/"   \
+        -i /etc/yum.conf
```

Comment:
It may, but I'm not entirely sure that `dnf` reads `yum.conf`.  It's probably safest/best to make this change in `/etc/dnf.conf`.

But why is this being done at all if we are replacing `mirrorlist`'d repos in any case?

Reply:
DNF on CentOS 7 uses /etc/yum.conf.

With out that change, yum/dnf takes a long time to check if the mirror list we are not using is current and will fail at this point if it can not reach the mirror list server.

Reply:
> DNF on CentOS 7 uses /etc/yum.conf.

I think it also uses `/etc/dnf.conf`.  Use of DNF on EL7 would be more consistent with EL8 I think.  EL8 doesn't still use `/etc/yum.conf` does it?